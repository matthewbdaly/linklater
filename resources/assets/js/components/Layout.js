import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import LinkList from './LinkList';
import LinkInput from './LinkInput';

export default class Layout extends Component {
    render() {
        return (
            <div>
                <LinkInput />
                <LinkList {...this.props} />
            </div>
        );
    }
}
