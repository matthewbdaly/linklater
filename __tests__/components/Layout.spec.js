import React from 'react';
import {findDOMNode} from 'react-dom';
import Layout from '../../resources/assets/js/components/Layout';
import renderer from 'react-test-renderer';

describe('Layout', () => {
  it('renders the layout', () => {
    const tree = renderer
      .create(<Layout filter='' links={[]} jwt='' graphql_route='' createUrl='' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
