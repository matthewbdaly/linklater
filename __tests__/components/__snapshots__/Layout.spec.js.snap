// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders the layout 1`] = `
<div>
  <form>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="url"
      >
        URL
      </label>
      <input
        autoComplete="off"
        className="form-control"
        name="url"
        placeholder="Page to save"
        required={true}
        type="url"
      />
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
    <p
      className="bookmarklet"
    >
      <a
        href="javascript:location.href='?url='+encodeURIComponent(location.href)"
      >
        LinkLater This!
      </a>
    </p>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id="linkfilter"
        onChange={[Function]}
        placeholder="Filter links..."
      />
      <button
        className="btn btn-primary"
        id="clear"
      >
        Clear filter
      </button>
    </div>
    <ul
      className="list-group"
      id="links"
    />
  </form>
</div>
`;
